lib/ansible/plugins/action/include_vars.py

```diff
@@ -152,6 +155,13 @@ def run(self, tmp=None, task_vars=None):
             scope[self.return_results_as_name] = results
             results = scope
 
+        if not self.override:
+            filtered = dict()
+            for (k,v) in results.items():
+                if k not in task_vars:
+                    filtered[k] = v
+            results = filtered
+
```

Comment:
this implementation seems to assume that `hash_merge=replace` if set to `merge` this would not create a result I would expect